ui/pages/confirmations/hooks/alerts/signatures/useDomainMismatchAlert.ts

```diff
@@ -0,0 +1,42 @@
+import { useMemo } from 'react';
+import { useSelector } from 'react-redux';
+import {
+  isValidSIWEOrigin,
+  WrappedSIWERequest,
+} from '@metamask/controller-utils';
+
+import { Alert } from '../../../../../ducks/confirm-alerts/confirm-alerts';
+import { Severity } from '../../../../../helpers/constants/design-system';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { currentConfirmationSelector } from '../../../../../selectors';
+
+import { SignatureRequestType } from '../../../types/confirm';
+import { isSIWESignatureRequest } from '../../../utils';
+
+export default function useDomainMismatchAlerts(): Alert[] {
+  const t = useI18nContext();
+
+  const currentConfirmation = useSelector(
+    currentConfirmationSelector,
+  ) as SignatureRequestType;
+  const { msgParams } = currentConfirmation;
+
+  const isValidSIWE = currentConfirmation?.msgParams && isSIWESignatureRequest(currentConfirmation);
```

Comment:
Do we need to verify `currentConfirmation?.msgParams` here if that is also being done inside `isSIWESignatureRequest` via the optional chaining?